generate-fit-profile/src/decode.rs

```diff
@@ -50,17 +50,33 @@ fn decode_function_def(msg: &MessageDefinition) -> TokenStream {
 
     quote! {
         #(#comments)*
-        fn #fn_name(mesg_num: MesgNum, data_map: &mut HashMap<u8, Value>, accumlators: &mut HashMap<u32, Value>, options: &HashSet<DecodeOption>) -> Result<Vec<FitDataField>> {
+        fn #fn_name(mesg_num: MesgNum, data_map: &mut HashMap<(u8, u8), Value>, accumlators: &mut HashMap<u32, Value>, developer_fields: &HashMap<(u8, u8), DeveloperFieldDescription>, options: &HashSet<DecodeOption>) -> Result<Vec<FitDataField>> {
             let mut fields = Vec::new();
-            let mut entries: VecDeque<(u8, Value)> = data_map.iter().map(|(k, v)| (*k, v.clone())).collect();
-            while let Some((def_num, value)) = entries.pop_front() {
-                match def_num {
-                    #(#match_arms)*
-                    _ => {
-                        if !options.contains(&DecodeOption::DropUnknownFields) {
-                            fields.push(unknown_field(def_num, value));
+            let mut entries: VecDeque<((u8, u8), Value)> = data_map.iter().map(|(k, v)| (*k, v.clone())).collect();
+            while let Some(((dev_data_idx, def_num), value)) = entries.pop_front() {
+                if dev_data_idx == 255 { // Not a developer data idx
+                    match def_num {
+                        #(#match_arms)*
+                        _ => {
+                            if !options.contains(&DecodeOption::DropUnknownFields) {
+                                fields.push(unknown_field(def_num, value));
+                            }
                         }
                     }
+                } else {
+                    let dev_definition = developer_fields
+                        .get(&(dev_data_idx, def_num))
+                        .expect("Developer fields must be defined before used.");
```

Comment:
Here we'd want to raise an ErrorKind variant like mentioned above as well, Probably named `MissingDeveloperDefinitionMessage`.

Reply:
added as requested